### ISSUE TITLE
User missing groups for shell script

### ISSUE THREAD

**Issue by ScottSWu** [jhuckaby/Cronicle#407]
## Summary

I have a `cronicle` user and `naser` group (which gives access to nas folders). The `cronicle` user's primary group is `cronicle`, and secondary group `naser`. However, when running the Shell Script Plugin (as user `cronicle`), the process only has the group `cronicle`. As a result, it is not able to access the nas folders. I also tried setting the primary group to `naser`,  but the group was still set to just `cronicle`.

An initial workaround might be setting the `gid` as well as `uid` in the plugin settings for jobs.

## Steps to reproduce the problem

Sample script:
```
#!/bin/sh
id cronicle
whoami
groups
bash -c "groups"
```

Sample output:
```
uid=1001(cronicle) gid=1002(cronicle) groups=1001(naser),1002(cronicle)
cronicle
cronicle
cronicle
```

### Operating system and version?

Ubuntu 20.04.1

### Node.js version?

v14.16.1

### Cronicle software version?

Version 0.8.56

**Comment by mikeTWC1984** [comment]
I guess that make sense. What you see is process uid/gid, when cronicle spawns a child process it only sets uid and gid. It sounds like it's possible to set secondary groups with process.getgroups() / process.setgroups()

**Comment by ScottSWu** [comment]
Hmm I see. `getgroups` / `setgroups` would need to be called in the spawned child process? rather than the parent.

Another workaround using `su` / `sudo`:
```
sudo -S -u cronicle /bin/sh -c "/nas/script.sh"
```

**Comment by mikeTWC1984** [comment]
Yeah, looks like it's impossible to assign secondary groups to child process, at least in node. I guess setgroups would only work for root user, which make no sense (can just run plugin as root). Probably you can run you plugin as "nas" user, or clone shellplug and set the clone to run as "nas" in case you want to run different events under different users.
